yearn/v2/strategies.py

```diff
@@ -96,8 +100,96 @@ def load_harvests(self):
+        latest_harvest_with_profit_current = harvests[latest_harvest_with_profit_idx]
+        second_latest_harvest_with_profit_current = harvests[latest_harvest_with_profit_idx + 1]
+        harvest_profit_current = latest_harvest_with_profit_current['event']['profit']
+        seconds_between_harvests_current = chain[latest_harvest_with_profit_current['block_number']]['timestamp'] - chain[second_latest_harvest_with_profit_current['block_number']]['timestamp']
+        block_before_profitable_harvest_current = latest_harvest_with_profit_current['block_number'] - 1
+        vault = self.vault.vault
+        strategy = self.strategy
+        strategy_debt_current = vault.strategies(strategy.address, block_identifier=block_before_profitable_harvest_current).dict()['totalDebt']
+        if strategy_debt_current == 0:
+            current_apr = 0
+        else:
+            current_apr = harvest_profit_current / strategy_debt_current * (SECONDS_IN_YEAR / seconds_between_harvests_current)    
+
+        # Second latest profitiable harvest
+        latest_harvest_with_profit_previous = harvests[second_latest_harvest_with_profit_idx]
+        second_latest_harvest_with_profit_previous = harvests[second_latest_harvest_with_profit_idx + 1]
+        harvest_profit_previous = latest_harvest_with_profit_previous['event']['profit']
+        seconds_between_harvests_previous = chain[latest_harvest_with_profit_previous['block_number']]['timestamp'] - chain[second_latest_harvest_with_profit_previous['block_number']]['timestamp']
+        block_before_profitable_harvest_previous = latest_harvest_with_profit_previous['block_number'] - 1
+        strategy_debt_previous = vault.strategies(strategy.address, block_identifier=block_before_profitable_harvest_previous).dict()['totalDebt']
+        if strategy_debt_previous == 0:
+            previous_apr = 0
+        else:
+            previous_apr = harvest_profit_previous / strategy_debt_previous * (SECONDS_IN_YEAR / seconds_between_harvests_previous)    
+
+        # Account for new vault use case
+        if previous_apr == 0 and current_apr > 0:
+            apr = current_apr
+        # Account for early withdrawal edge case
+        elif current_apr > previous_apr * 2  or current_apr * 2 < previous_apr:
+            apr = previous_apr
+        # Account for sunshine use case
+        else:
+            apr = current_apr
+
+        # Determine fees
+        strategy_performance_fee = vault.strategies(strategy.address).dict()['performanceFee'] / 1e4
+        vault_performance_fee = vault.performanceFee() / 1e4
+        performance_fee = vault_performance_fee + strategy_performance_fee
+        management_fee = vault.managementFee() / 1e4 if hasattr(vault, "managementFee") else 0
+        
+        # Subtract fees
+        apr_minus_fees = apr * (1 - performance_fee) - management_fee
+        
+        # Convert APR to annualized estimated APY
+        net_apy = (1 + (apr_minus_fees / 365)) ** 365 - 1
```

Comment:
Not trying to be pedantic here, I'm just wondering why we're using here 365 if the `SECONDS_IN_YEAR` constant uses 365.25 as a base. Maybe it's not an issue and a common economic agreement to use 365 for apy calculations but it looks slightly off and both values should either use 365 or 365.25 imo.

Reply:
Alternatively, we could instead use daily for fantom, and weekly for Ethereum. But I'm not quite sure how to do this in exporter; is there a way to read what network a contract is on?

Reply:
@dudesahn the exporter is running within a given brownie network that you provide on startup. So basically all contracts seen within one exporter process are scoped to this one network. As you already said the current network can be checked with `chain.id` and we have some constants in `yearn/networks.py` which could be used like so:

https://github.com/yearn/yearn-exporter/blob/master/yearn/prices/magic.py#L42